test/unit/attachment_test.rb

```diff
@@ -58,11 +58,16 @@ class AttachmentTest < ActiveSupport::TestCase
     test "should be invalid when the command paper number starts with '#{prefix}'" do
       attachment = build(:file_attachment, command_paper_number: "#{prefix} 1234")
       assert_not attachment.valid?
-      expected_message = "is invalid. The number must start with one of #{Attachment::VALID_COMMAND_PAPER_NUMBER_PREFIXES.join(', ')}"
+      expected_message = "is invalid. The number must start with one of #{Attachment::VALID_COMMAND_PAPER_NUMBER_PREFIXES.join(', ')}, followed by a space"
       assert attachment.errors[:command_paper_number].include?(expected_message)
     end
   end
 
+  test "should be invalid when the command paper number has no space after the prefix" do
+    attachment = build(:file_attachment, command_paper_number: "C1234")
```

Comment:
To avoid hard-coding, I could change this to:

```suggestion
    attachment = build(:file_attachment, command_paper_number: "#{Attachment::VALID_COMMAND_PAPER_NUMBER_PREFIXES.first}1234")
```

But it feels quite heavy handed and less readable. I invite feedback!

Reply:
I agree it's less readable. I'd say hard coding like this is acceptable for a test, I suppose it depends on how likely the prefixes are to change. Also C1234 is an invalid prefix, not only because the lack of space but the lack of a period no? I accept this is the nit pickiest thing in the universe 😄

Reply:
Thanks - updated.